File: src/main/java/com/example/socialnetworkingapp/model/bio/BioService.java
package com.example.socialnetworkingapp.model.bio;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor
public class BioService {

    private final BioRepository bioRepository;

    public Bio addBio(Bio bio) {
        return this.bioRepository.save(bio);
    }

    public Bio updateBio(Bio bio) {
        return this.bioRepository.save(bio);
    }

    public void deleteBio(Bio bio) {
        this.bioRepository.delete(bio);
    }

    public void deleteBioById(Long bioId) {
        this.bioRepository.deleteById(bioId);
    }
}

File: socialnetworkingapp-front/src/app/bio/bio.service.ts
import { Injectable } from '@angular/core';
import {Bio} from "./bio";
import {AccountService} from "../account/account.service";
import {Account} from "../account/account";
import {HttpClient} from "@angular/common/http";
import {environment} from "../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class BioService {

  private accountService: AccountService;
  private account: Account;

  constructor(private http: HttpClient) { }

  addBio(bio: Bio) {
    return this.http.post<Bio>(`${environment.apiBaseUrl}/bio/add`, bio);
  }

  getBioByAccountId(id: number) {
    return this.http.get<Bio>(`${environment.apiBaseUrl}/bio/acc_ref/${id}`)
  }

  deleteBioByAccountId(id: number) {
    return this.http.delete<void>(`${environment.apiBaseUrl}/bio/delete/${id}`);
  }

  updateBio(bio: Bio) {
    return this.http.put<Bio>(`${environment.apiBaseUrl}/bio/update`, bio);
  }

  deleteBioById(bioId: number) {
    return this.http.delete<void>(`${environment.apiBaseUrl}/bio/delete/${bioId}`);
  }
}
